custom/enikshay/management/commands/data_dumps_person_case.py

```diff
@@ -105,6 +119,39 @@ def get_all_episode_cases_from_person(domain, person_case_id):
     ]
 
 
+def get_all_episode_cases_from_occurrence(domain, occurrence_case_id):
+    return CaseAccessors(domain).get_reverse_indexed_cases(
+        [occurrence_case_id], case_types=[CASE_TYPE_EPISODE])
+
+
+def get_last_occurrence(person_case):
+    occurrence_cases = get_all_occurrence_cases_from_person(person_case.domain, person_case.case_id)
+    open_occurrence_cases = [
+        occurrence_case for occurrence_case in occurrence_cases
+        if not occurrence_case.closed
+    ]
+    if len(open_occurrence_cases) > 1:
+        raise Exception("Multiple open occurrence cases for person %s" % person_case.case_id)
+    elif len(open_occurrence_cases) == 1:
+        return open_occurrence_cases[0]
+    else:
+        # dict to maintain episodes ids to occurrence case mapping to avoid fetching occurrence again
+        # later from latest episode case
+        episode_id_to_occurrence = {}
+        all_episode_cases = []
+        for occurrence_case in occurrence_cases:
+            episodes = get_all_episode_cases_from_occurrence(occurrence_case.domain, occurrence_case.case_id)
```

Comment:
nit: you can pass the list of occurrence case ids to `get_reverse_indexed_cases` to fetch all the episodes in one db call.

Reply:
oh, I guess you need to know which occurrence it came from, so you'd have to split that back out anyways, maybe not worth it.